carbon-design-system/carbon
# Update storybook to load flags conditionally and improve build perf

Updates our storybook setup with the following enhancements:

- The `enable-css-custom-properties` flag is off by default so that our stable storybook previews aren't broken
- You can enable this flag through the `CARBON_REACT_STORYBOOK_USE_CUSTOM_PROPERTIES` flag
- Updated `sass-loader` for the function-based `prependData` to conditionally load in flags for our CSS compilation
- Update how we bring in styles to include build performance

#### Changelog

**New**

**Changed**

- Update `sass-loader` dependency for react storybook
- Update storybook config to load styles through JS with `prependData` function
- Update setup to conditionally load custom properties

**Removed**

#### Testing / Reviewing

- Make sure to reinstall dependencies as `sass-loader` has been updated
- Run `yarn storybook` and verify that:
  - [ ] The carbon theme panel is not visible
  - [ ] The carbon type panel is not visible
  - [ ] Styles load as expected
  - [ ] Make an edit in `_accordion.scss` and verify change occurs in a couple of seconds
- Run `CARBON_REACT_STORYBOOK_USE_CUSTOM_PROPERTIES=true yarn storybook` and verify that:
  - [ ] The carbon theme panel is visible
  - [ ] The carbon type panel is visible
  - [ ] Styles load as expected
  - [ ] Updating a value in the carbon theme panel updates the current story
  - [ ] Updating a value in the carbon type panel updates the corresponding type token in the story
